examples/caa_sycophancy.ipynb

```diff
@@ -0,0 +1,458 @@
+    "    read_token_index=-2,\n",
+    "    show_progress=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(steering_vector)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's sanity check our vector by evaluating the cosine similarity with the ground truth sycophancy vector."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Download the CAA sycophancy vectors for layer 15\n",
+    "!wget https://raw.githubusercontent.com/nrimsky/CAA/main/vectors/vec_layer_15_Llama-2-7b-chat-hf.pt\n",
+    "!wget https://raw.githubusercontent.com/nrimsky/CAA/main/vectors/vec_layer_15_Llama-2-13b-chat-hf.pt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch.nn.functional import cosine_similarity\n",
+    "\n",
+    "original_steering_vector = torch.load(f'vec_layer_15_Llama-2-{model_size}-chat-hf.pt')\n",
+    "our_steering_vector = steering_vector.layer_activations[15]\n",
+    "print(f\"Cosine similarity: {cosine_similarity(original_steering_vector, our_steering_vector, dim=0):.3f}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Steer with Steering Vectors"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can apply steering vectors with `SteeringVector.apply` as follows: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for multiplier in (-1, 0, 1):\n",
+    "    with steering_vector.apply(model, multiplier=multiplier, min_token_index=0):\n",
+    "        # Within the scope, model activations are modified\n",
+    "        result = evaluate_model(model, tokenizer, test_dataset)\n",
+    "        print(f\"{multiplier} steered model: {result:.3f}\")\n",
+    "        # Upon leaving the scope, original model activations are restored"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Note: PyTorch Hooks\n",
+    "To apply steering vectors to the model, we must have some way of modifying the model's forward pass. The approach taken in the official codebases for Representation Engineering and Contrastive Activation Addition both do this by wrapping the model's blocks. This approach is conceptually simpler but has notable limitations, e.g. being unable to work for other models\n",
+    " \n",
+    "The `steering_vectors` library uses Pytorch hooks instead of model wrappers to modify the underlying model's forward pass. You can read more about hooks at the [official PyTorch documentation](https://pytorch.org/docs/stable/generated/torch.nn.modules.module.register_module_forward_hook.html). "
```

Comment:
nit: I still don't think this cell belongs here, we can put this into the `about` page of the docs, since you don't actually need to know about PyTorch hooks in order to use this library. Arguably, not needing to care about things like hooks / model wrapping is the selling point of the library, in the same way that the selling point of PyTorch is not needing to know/care about CUDA programming. This would be like the Pytorch docs for `cosine_similarity` linking you to some CUDA kernel code, which for most users would be confusing and/or terrifying.